core/types/transaction_marshalling.go

```diff
@@ -94,6 +104,35 @@ func (t *Transaction) MarshalJSON() ([]byte, error) {
 		enc.V = (*hexutil.Big)(tx.V)
 		enc.R = (*hexutil.Big)(tx.R)
 		enc.S = (*hexutil.Big)(tx.S)
+	case *ShutterTx:
+		enc.ChainID = (*hexutil.Big)(tx.ChainID)
+		enc.Nonce = (*hexutil.Uint64)(&tx.Nonce)
+		enc.Gas = (*hexutil.Uint64)(&tx.Gas)
+		enc.MaxFeePerGas = (*hexutil.Big)(tx.GasFeeCap)
+		enc.MaxPriorityFeePerGas = (*hexutil.Big)(tx.GasTipCap)
+		enc.EncryptedPayload = (*hexutil.Bytes)(&tx.EncryptedPayload)
+		enc.L1BlockNumber = (*hexutil.Uint64)(&tx.L1BlockNumber)
+		enc.BatchIndex = (*hexutil.Uint64)(&tx.BatchIndex)
+		enc.To = t.To()
+		enc.V = (*hexutil.Big)(tx.V)
+		enc.R = (*hexutil.Big)(tx.R)
+		enc.S = (*hexutil.Big)(tx.S)
+	case *BatchTx:
+		enc.ChainID = (*hexutil.Big)(tx.ChainID)
+		if tx.Transactions != nil {
+			enc.Transactions = make([]hexutil.Bytes, len(tx.Transactions))
+			for k, v := range tx.Transactions {
+				enc.Transactions[k] = hexutil.Bytes(v)
+			}
+		}
+		enc.Timestamp = (*hexutil.Big)(tx.Timestamp)
+		enc.DecryptionKey = (*hexutil.Bytes)(&tx.DecryptionKey)
+		enc.L1BlockNumber = (*hexutil.Uint64)(&tx.L1BlockNumber)
+		enc.BatchIndex = (*hexutil.Uint64)(&tx.BatchIndex)
+		enc.To = t.To()
```

Comment:
`BatchTxs` don't have a `To` field

Reply:
I would leave this in, since it returns `nil` anyways and it is consistent with the other tx types
(it is not "omitempty" tagged, so it will be encoded as "null" anyways).

Reply:
hmm, not convinced. Then we should arguably include our fields in all other types then as well. But it's also just a nitpick, so I'm fine to leave it as is.